### PR TITLE
Added a function "search" to look for a signature in memory. Ex: chip…

### PR DESCRIPTION
**Pull request by hcwang1126**
it's convenient for a BIOS engineer to dump some resource tables.

### CODE REVIEW

File: chipsec/utilcmd/mem_cmd.py [hunk 1]
```diff
@@ -128,6 +129,17 @@ def run(self):
...
 
+        elif 'search' == op and len(self.argv) > 5:
+            phys_address = int(self.argv[3],16)
+            size         = int(self.argv[4],16)
+            target = 0
+            for i in range(0, len(self.argv[5])):
+                target += ord(self.argv[5][i]) << (i * 8)
+                      
+            buffer = self.cs.mem.read_physical_mem( phys_address, size )
+            offset = buffer.find(self.argv[5])
```
Comment (ErikBjorge) [inline chipsec/utilcmd/mem_cmd.py]: Should handle the case where the search string is not found and -1 is returned.

**Comment by ErikBjorge** [comment]
Other than the one small change I think this works well.  Used it to manually find the ACPI anchor and table signatures.

File: chipsec/utilcmd/mem_cmd.py (hunk 1 above)
Comment (hcwang1126) [reply chipsec/utilcmd/mem_cmd.py]: Agree.

**Comment by hcwang1126** [comment]
Already commit a fix to handle that the signature is not found.

**Pull request merged**